### PR TITLE
feat(shopify-checkout): add in shopify scripts support

### PR DESCRIPTION
**Pull request by alexcasche**
### WHY are these changes introduced?

Adds [[LS-1548](https://nacelle.atlassian.net/browse/LS-1548)] 

### WHAT is this pull request doing?

- Adds support for `discountApplications` and `lineItems` to return all Shopify Scripts related information
- Locks the package to specified version of Shopify Storefront API

**Note:** The approach here was highly dictated by the upcoming rewrite of `shopify-checkout` outlined [here](https://nacelle.atlassian.net/wiki/spaces/DD/pages/1667301474/nacelle+shopify-checkout+version+1.0.0#Implementation-Strategy).

### HOW to test?

`git checkout LS-1548--line-item-mutations-queries`
`npm run test:ci`

### NOTES

Looks like the `next-starter` is failing to build on Vercel.  Will create a separate Jira ticket for this

### CODE REVIEW

File: packages/shopify-checkout/src/utils/buildCheckout.ts [hunk 1]
```diff
@@ -2,11 +2,16 @@ import { BuildCheckoutParams, ShopifyCheckout } from '../checkout-client.types';
...
+  lineItems,
+  discountApplications
 }: BuildCheckoutParams): ShopifyCheckout {
   return {
     id,
     url: webUrl,
-    completed: false
+    completed: Boolean(completedAt),
+    lines: lineItems.edges.map((edge) => edge.node),
+    discounts: discountApplications.edges.map((edge) => edge.node)
```
Comment (jasonredick) [inline packages/shopify-checkout/src/utils/buildCheckout.ts:15]: Could we write tests for these lines? This file has pretty low coverage (probably cause it's so small) and running the test points to these two lines.

File: packages/shopify-checkout/src/utils/createGqlClient.ts [hunk 2]
```diff
@@ -41,18 +41,21 @@ export default function createGqlClient({
...
-  storefrontApiVersion,
   storefrontCheckoutToken
 }: CreateGqlClientParams): GqlClient {
   const gqlClient: GqlClient = ({ query, variables }) => {
-    let endpoint = customEndpoint;
+    let endpoint = customEndpoint || '';
+
+    if (!storefrontCheckoutToken) {
+      throw new Error(missingAccessTokenMessage);
+    }
```
Comment (jasonredick) [inline packages/shopify-checkout/src/utils/createGqlClient.ts:51]: We should probably test for this error too.

**Review by jasonredick — APPROVED** [review]
This looks good to me. There are some improvements to be made in our tests, but I don't know if it's enough to hold this up.

**Review by NWRichmond — APPROVED** [review]
Thank you for the coverage improvements in https://github.com/getnacelle/nacelle-js/pull/84/commits/487d86b8a0c32743cd2923ac93e9e7f835ff4d26.

In local testing, I found that Shopify Scripts discounts were returned in `cart.lines[x].discountAllocations` and `cart.discounts`. Excellent!

**Pull request merged**